### PR TITLE
WIP : Implement forum in chat

### PR DESCRIPTION
**Pull request by cesar-cardinale**
Add the possibility to show a forum in the chat.

### CODE REVIEW

File: front/chat/src/Components/Chat.svelte [hunk 1]
```diff
@@ -195,6 +200,16 @@
                     />
                 {/if}
 
+                <ChatForumRooms
+                    {showForums}
+                    searchValue={searchValue.toLocaleLowerCase()}
+                    on:activeThread={handleActiveThread}
+                    on:showForums={handleShowForums}
```
Comment (gparant) [inline front/chat/src/Components/Chat.svelte:207]: Why don't use this directly in the component `ChatForumRooms` and have a handler and props?

File: back/src/Model/GameRoom.ts [hunk 2]
```diff
@@ -69,6 +69,7 @@ export class GameRoom implements BrothersFinder {
     private nextUserId = 1;
 
     private roomListeners: Set<RoomSocket> = new Set<RoomSocket>();
+    mucManager: MucManager | null = null;
```
Comment (moufmouf) [inline back/src/Model/GameRoom.ts:72]:
```suggestion
```

File: back/src/Services/MucManager.ts [hunk 3]
```diff
@@ -23,69 +25,103 @@ export class MucManager {
...
     }
 
     public async init(mapDetails: MapDetailsData) {
-        const allMucRooms = await ejabberdClient.getAllMucRooms();
-        const allMucRoomsOfWorld: string[] = [];
-        if (Axios.isAxiosError(allMucRooms)) {
-            console.warn("Error to get allMucRooms (AxiosError) : ", allMucRooms.response);
-        } else if (allMucRooms instanceof Error) {
-            console.warn("Error to get allMucRooms : ", allMucRooms);
+        const allMucRoomsCreated = await ejabberdClient.getAllMucRooms();
```
Comment (moufmouf) [inline back/src/Services/MucManager.ts:61]: This will not scale if the system has too many rooms.

Comment (moufmouf) [reply back/src/Services/MucManager.ts:61]: Maybe we could "cache" the list of live zones and remove the live zones that are not in the cached list every hour.

This way, we make only one request to `getAllMucRooms` per hour?

File: docker-compose.yaml [hunk 4]
```diff
@@ -251,6 +251,7 @@ services:
 
   messages:
     image: thecodingmachine/workadventure-back-base:v1-16
+    platform: linux/amd64
```
Comment (moufmouf) [inline docker-compose.yaml:254]:
```suggestion
    # Remove this when node-grpc module is updated to the version that supports ARM64
    platform: linux/amd64
```

File: libs/map-editor-types/yarn.lock [hunk 5]
```diff
@@ -0,0 +1,4 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
+# yarn lockfile v1
+
+
```
Comment (moufmouf) [inline libs/map-editor-types/yarn.lock:4]: To delete

File: messages/ts-proto-nest-generated/google/protobuf/wrappers.ts [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+}
+
+export const GOOGLE_PROTOBUF_PACKAGE_NAME = "google.protobuf";
+
+// If you get a compile-error about 'Constructor<Long> and ... have no overlap',
+// add '--ts_proto_opt=esModuleInterop=true' as a flag when calling 'protoc'.
+if (util.Long !== Long) {
+    util.Long = Long as any;
+    configure();
+}
```
Comment (moufmouf) [inline messages/ts-proto-nest-generated/google/protobuf/wrappers.ts:104]: To delete (with all the files in ts-proto-nest-generated)

File: pusher/src/Controller/IoSocketController.ts [hunk 7]
```diff
@@ -694,7 +695,7 @@ export class IoSocketController {
         client.token = ws.token;
         client.batchedMessages = new BatchMessage();
         client.batchTimeout = null;
-        client.emitInBatch = (payload: SubMessage): void => {
+        client.emitInBatch = (payload: SubMessage | SubChatMessage): void => {
```
Comment (moufmouf) [inline pusher/src/Controller/IoSocketController.ts:698]: The whole initClient method is just plain wrong.

Idea:

Out of initClient, add emitInBatch or emitInBatchChat

Validate the client with Zod (with 2 zod types): ExSocketInterface and ExChatSocketInterface

**Pull request closed without merging**